apps/ios/SimpleX NSE/NotificationService.swift

```diff
@@ -326,7 +332,13 @@ class NotificationService: UNNotificationServiceExtension {
         if let t = threadId {
             threadId = nil
             if NSEThreads.shared.endThread(t) {
-                suspendChat(nseSuspendTimeout)
+                logger.debug("NotificationService.deliverBestAttemptNtf: will suspend")
+                DispatchQueue.global().asyncAfter(deadline: .now() + nseSuspendDelay) {
+                    if NSEThreads.shared.noThreads {
+                        logger.debug("NotificationService.deliverBestAttemptNtf: suspending...")
+                        suspendChat(nseSuspendTimeout)
+                    }
+                }
```

Comment:
could check noThreads first, then delay and check again. It's probably unnecessary to delay if there's already noThreads

Reply:
add comment